### PR TITLE
[Enhancement] Hang on show routine load stmt

### PR DESCRIPTION
**Pull request by rickif**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #https://github.com/StarRocks/starrocks/issues/9778

## Problem Summary(Required) ：

Sometimes, Kafka's response may be slow. When the routine load job is scheduled, the partition info needs to be checked. The previous implementation gets Kafka partition info with the write lock, while the show routine load statement needs read lock,  which may lead to the hang of the statement. This PR introduces two modifications:
1. getting Kafka info with a read lock.
2. some refactoring to simplify the code.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

**Comment by rickif** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 24 / 39 (61.54%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/routineload/KafkaRoutineLoadJob.java|9|23|39.13%|[297, 298, 299, 300, 301, 302, 303, 307, 311, 313, 322, 323, 324, 325]|
|:large_blue_circle:|com/starrocks/load/routineload/RoutineLoadJob.java|15|16|93.75%|[1241]|

**Review by meegoo — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/load/routineload/KafkaRoutineLoadJob.java [hunk 1]
```diff
@@ -279,64 +279,55 @@ protected void unprotectUpdateProgress() {
...
+                // It may be slow to get info from kafka.
+                // It's ok to hold read lock here.
+                newCurrentKafkaPartition = getAllKafkaPartitions();
+            } catch (Exception e) {
+                String msg = "Job failed to fetch all current partition with error [" + e.getMessage() + "]";
+                LOG.warn(new LogBuilder(LogKey.ROUTINE_LOAD_JOB, id)
+                        .add("error_msg", msg)
+                        .build(), e);
+                if (this.state == JobState.NEED_SCHEDULE) {
+                    unprotectUpdateState(JobState.PAUSED,
```
Comment (trueeyu) [inline fe/fe-core/src/main/java/com/starrocks/load/routineload/KafkaRoutineLoadJob.java:303]: May be unsafe to update with read lock?

Comment (rickif) [reply fe/fe-core/src/main/java/com/starrocks/load/routineload/KafkaRoutineLoadJob.java:308]: Updated.

**Review by chaoyli — APPROVED** [review]

**Comment by rickif** [comment]
run starrocks_fe_unittest

**Comment by rickif** [comment]
@mergifyio rebase

**Comment by rickif** [comment]
run starrocks_fe_unittest

**Comment by rickif** [comment]
@Mergifyio rebase

**Pull request closed without merging**